### PR TITLE
HPCC-12683 Use interleaved memory policy

### PR DESCRIPTION
**Pull request by mckellyln**
Added FindNUMA.cmake for developers to notice numa package requirement at devel/build time.

Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

(recreated branch from an incorrect rebase)

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/roxiemem/roxiemem.cpp [hunk 1]
```diff
@@ -169,6 +173,28 @@ static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages,
...
     heapUseHugePages = false;
 
+#ifdef _USE_NUMA
+    bool has_numa = numa_available() >= 0;
+    if (has_numa)
+    {
+        struct bitmask *nodemask = numa_get_mems_allowed();
+        if (nodemask && *nodemask->maskp)
+        {
+            int serr = set_mempolicy(MPOL_INTERLEAVE, nodemask->maskp, nodemask->size);
```
Comment (ghalliday) [inline roxie/roxiemem/roxiemem.cpp]: Should it be possible to configure this via an option - even if it is only to easily test what effect it has?

**Comment by ghalliday** [comment]
Any information about how it changes performance?

**Comment by ghalliday** [comment]
(The option doesn't need to be passed into the function, it could be a separate setting - although it is getting close to the point where we should be passing in a structure of options...)

**Comment by ghalliday** [comment]
@mckellyln I think it is worth making the option softer.  Any idea how performance changes?

**Comment by mckellyln** [comment]
I will close and reconsider for 5.4 with changes for making it user / system configurable.

**Pull request closed without merging**